File: app/src/main/java/com/saitejajanjirala/imdbclone/db/IdsToStringConverter.java
package com.saitejajanjirala.imdbclone.db;

import androidx.room.TypeConverter;

import java.util.ArrayList;
import java.util.List;

public class IdsToStringConverter {

    @TypeConverter
    public String listToStringConverter(List<Integer> ids) {
        String val = "";
        for (int i : ids) {
            val += i + ",";
        }
        if (val.equals("")) {
            return val;
        }
        return val.substring(0, val.length() - 1);

    }

    @TypeConverter
    public List<Integer>  stringToList(String x){
        List<Integer> list=new ArrayList<>();
        if(x.length()==0) return list;
        else if(x.length()==1) {
            list.add(Integer.parseInt(x));
        }
        else {
            for (String i : x.split(",")) {
                list.add(Integer.parseInt(i));
            }
        }
        return list;
    }
}

File: app/src/main/java/com/saitejajanjirala/imdbclonekotlin/db/IdsToStringConverter.kt
package com.saitejajanjirala.imdbclonekotlin.db

import androidx.room.TypeConverter
import java.util.ArrayList

class IdsToStringConverter {
    @TypeConverter
    fun listToStringConverter(ids: List<Int>): String {
        var str = ""
        for (i in ids) {
            str += "$i,"
        }
        return if (str == "") {
            str
        } else str.substring(0, str.length - 1)
    }

    @TypeConverter
    fun stringToList(x: String): List<Int> {
        val list: MutableList<Int> = ArrayList()
        when {
            x.isEmpty() -> return list
            x.length == 1 -> {
                list.add(x.toInt())
            }
            else -> {
                for (i in x.split(",").toTypedArray()) {
                    list.add(i.toInt())
                }
            }
        }
        return list
    }
}